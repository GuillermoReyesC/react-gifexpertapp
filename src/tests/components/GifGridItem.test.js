import React from 'react'
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'  

describe('Pruebas en <GifGridItem/ >', () => {


    const title= 'titulo';
    const url= 'http://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem  title={ title } url={ url } />)

    
    
    test('Debe de mostrarse el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot( );
        
        
    })


    test('debe de mostrar data enviada por propTypes ', () => {

    
        const wrapper = shallow(
        
            <GifGridItem
                title={ title }
                url={ url }

        
            />) 
               
        
        expect( wrapper ).toMatchSnapshot()    
            
    });


    test('debe tener un parrafo con un texto', () => {

        
        
        const parrafoTexto = wrapper.find('p').text()

        expect( parrafoTexto.trim() ).toBe( title );
    
        


    });

    test('debe tener la iomagen igual a la url y alt de los props', () => {

        const img = wrapper.find('img')

        //console.log(img.props().alt)

        expect(img.prop('src') ).toBe(url)
        expect(img.prop('alt') ).toBe(title)
        
    });

    test('el <div> debe tener animate_fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className')
        // animate_fadeIn

        console.log(div.prop('className'))

        expect(div.prop('className')).toEqual('card animate__animated animate__fadeIn')

        expect( className.includes('animate__fadeIn') ).toBe(true)
        
    });


});