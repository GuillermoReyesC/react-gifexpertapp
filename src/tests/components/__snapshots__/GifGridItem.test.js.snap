// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/ > Debe de mostrarse el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
     
    titulo
  </p>
</div>
`;

exports[`Pruebas en <GifGridItem/ > debe de mostrar data enviada por propTypes  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
     
    titulo
  </p>
</div>
`;
